pkg/dns/serviceresolver.go

```diff
@@ -210,14 +207,6 @@ func (b *ServiceResolver) Records(dnsName string, exact bool) ([]msg.Service, er
 			return nil, errNoSuchName
 		}
 
-		hostnameMappings := noHostnameMappings
```

Comment:
I talked to @smarterclayton about this. IIRC they moved to a field. Regardless, this is correct.